LDAP login failure with 4.87.0

## Issue

Add these issues to elsewhere:
- Snap: https://github.com/wekan/wekan-snap/issues
- LDAP: https://github.com/wekan/wekan-ldap/issues
- UCS: https://github.com/wekan/univention/issues

Other Wekan issues can be added here.

**Server Setup Information**:

* Note: Please anonymize info, and do not add to this public issue any of your Wekan board URLs, passwords, API tokens etc, do you understand?:
* Did you test in newest Wekan?: Yes
* For new Wekan install, did you configure root-url correctly so Wekan cards open correctly https://github.com/wekan/wekan/wiki/Settings ? Yes
* Wekan version: 4.87.0
* If this is about old version of Wekan, what upgrade problem you have?:
* Operating System: Linux
* Deployment Method(snap/docker/sandstorm/mongodb bundle/source):snap
* Http frontend if any (Caddy, Nginx, Apache, see config examples from Wekan GitHub wiki first):Nginx
* Node Version: 12.20.1
* MongoDB Version: 3.2.22
* Wekan only works on newest desktop Firefox/Chromium/Chrome/Edge/Chromium Edge and mobile Chrome. What webbrowser version are you using?Chrome 87.0.4280.141

**Problem description**:
- *REQUIRED: Unable to login via LDAP
2021-01-27T02:03:34Z wekan.wekan[63755]: [Symbol(kCapture)]: false
2021-01-27T02:03:34Z wekan.wekan[63755]: },
2021-01-27T02:03:34Z wekan.wekan[63755]: tlsOptions: { rejectUnauthorized: true }
2021-01-27T02:03:34Z wekan.wekan[63755]: }
2021-01-27T02:03:44Z wekan.wekan[63755]: [ERROR] connection time out 10000
2021-01-27T02:03:44Z wekan.wekan[63755]: [ERROR] Error: Timeout 

- *Explain steps how to reproduce* - Login page appears, attempt login and fails.
- *In webbrowser, what does show Right Click / Inspect / Console ? Chrome shows more detailed info than Firefox.*
- *If using Snap, what does show command `sudo snap logs wekan.wekan` ? Please anonymize logs.* 
- *If using Docker, what does show command `sudo docker logs wekan-app` ? Please anonymize logs.*
- *If logs are very long, attach them in .zip file*

Hi there,
I hope I am right here.
I think I have the same Issue as mentioned above with Version 4.87 - my ldap authentication worked fine in 4.86 but yesterday an user reported, that it doesn't work anymore, the system updated 2 days ago.
refresh-date: 2 days ago, at 02:12 CET
```
channels:
  latest/stable:    4.87            2021-01-26 (1296) 171MB -
  latest/candidate: 4.87            2021-01-26 (1296) 171MB -
  latest/beta:      4.87            2021-01-26 (1296) 171MB -
  latest/edge:      4.87-4-g4bd3329 2021-01-26 (1299) 171MB -
installed:          4.87                       (1296) 171MB -
```

Wekan version: 4.87
Operating System: Ubuntu 18.04
Deployment Method: snap
Http frontend if any: Nginx
What webbrowser version are you using?: Chrome 88.0.4324.96

Already tried to set rejectUnauthorized to false with `sudo snap set wekan ldap-reject-unauthorized='false'` but it still logs `tlsOptions: { rejectUnauthorized: true }`.

Debug Information for an login attempt:
```
2021-01-28T08:37:39Z wekan.wekan[40075]: [INFO] Connecting "ldaps://10.71.0.33:636"
2021-01-28T08:37:39Z wekan.wekan[40075]: [DEBUG] connectionOptions{
2021-01-28T08:37:39Z wekan.wekan[40075]:   url: 'ldaps://10.71.0.33:636',
2021-01-28T08:37:39Z wekan.wekan[40075]:   timeout: 10000,
2021-01-28T08:37:39Z wekan.wekan[40075]:   connectTimeout: 10000,
2021-01-28T08:37:39Z wekan.wekan[40075]:   idleTimeout: 10000,
2021-01-28T08:37:39Z wekan.wekan[40075]:   reconnect: true,
2021-01-28T08:37:39Z wekan.wekan[40075]:   log: Logger {
2021-01-28T08:37:39Z wekan.wekan[40075]:     _events: [Object: null prototype] {},
2021-01-28T08:37:39Z wekan.wekan[40075]:     _eventsCount: 0,
2021-01-28T08:37:39Z wekan.wekan[40075]:     _maxListeners: undefined,
2021-01-28T08:37:39Z wekan.wekan[40075]:     _level: 30,
2021-01-28T08:37:39Z wekan.wekan[40075]:     streams: [ [Object] ],
2021-01-28T08:37:39Z wekan.wekan[40075]:     serializers: null,
2021-01-28T08:37:39Z wekan.wekan[40075]:     src: false,
2021-01-28T08:37:39Z wekan.wekan[40075]:     fields: {
2021-01-28T08:37:39Z wekan.wekan[40075]:       name: 'ldapjs',
2021-01-28T08:37:39Z wekan.wekan[40075]:       component: 'client',
2021-01-28T08:37:39Z wekan.wekan[40075]:       hostname: 'srvrecendtubuntu01',
2021-01-28T08:37:39Z wekan.wekan[40075]:       pid: 41149
2021-01-28T08:37:39Z wekan.wekan[40075]:     },
2021-01-28T08:37:39Z wekan.wekan[40075]:     [Symbol(kCapture)]: false
2021-01-28T08:37:39Z wekan.wekan[40075]:   },
2021-01-28T08:37:39Z wekan.wekan[40075]:   tlsOptions: { rejectUnauthorized: true }
2021-01-28T08:37:39Z wekan.wekan[40075]: }
2021-01-28T08:37:49Z wekan.wekan[40075]: [ERROR] connection time out 10000
2021-01-28T08:37:49Z wekan.wekan[40075]: [ERROR] Error: Timeout
```

LDAP Config:
```
root@xxx:/# snap get wekan |grep ldap
default-authentication-method                     ldap
ldap-authentication                               true
ldap-authentication-password                      xxx
ldap-authentication-userdn                        xxx@xxx.xxx
ldap-background-sync                              true
ldap-background-sync-import-new-users             true
ldap-background-sync-interval                     100
ldap-background-sync-keep-existant-users-updated  true
ldap-base                                         OU=xxx,DC=tcl,DC=xxx
ldap-basedn                                       OU=xxx,DC=tcl,DC=xxx
ldap-enable                                       true
ldap-encryption                                   ssl
ldap-fullname-field                               CN
ldap-host                                         10.71.0.33
ldap-log-enabled                                  true
ldap-port                                         636
ldap-reject-unauthorized                          false
ldap-user-search-field                            sAMAccountName
ldap-user-search-filter                           (memberof:1.2.840.113556.1.4.1941:=CN=wekan_enabled,OU=xxx,OU=xxx,DC=tcl,DC=xxx)
ldap-username-field                               sAMAccountName
```

Not sure if it has to do something with this problem, but in logs I found the following Exceptions:

```
2021-01-27T12:52:50Z wekan.wekan[237285]: [INFO] Enabling LDAP Background Sync
2021-01-27T12:52:50Z wekan.wekan[237285]: Exception in callback of async function: TypeError: str.toLowerCase is not a function
2021-01-27T12:52:50Z wekan.wekan[237285]:     at Object.later.parse.text (/snap/wekan/1296/programs/server/npm/node_modules/meteor/percolate_synced-cron/node_modules/later/later.js:1506:34)
2021-01-27T12:52:50Z wekan.wekan[237285]:     at Object.schedule (packages/wekan-ldap/server/sync.js:452:22)
2021-01-27T12:52:50Z wekan.wekan[237285]:     at scheduleEntry (packages/percolate_synced-cron.js:126:24)
2021-01-27T12:52:50Z wekan.wekan[237285]:     at packages/percolate_synced-cron.js:163:7
2021-01-27T12:52:50Z wekan.wekan[237285]:     at Function._.each._.forEach (packages/underscore.js:147:22)
2021-01-27T12:52:50Z wekan.wekan[237285]:     at packages/percolate_synced-cron.js:162:7
2021-01-27T12:52:50Z wekan.wekan[237285]:     at Object.startup (packages/meteor.js:992:5)
2021-01-27T12:52:50Z wekan.wekan[237285]:     at Object.SyncedCron.start (packages/percolate_synced-cron.js:160:10)
2021-01-27T12:52:50Z wekan.wekan[237285]:     at addCronJobDebounced (packages/wekan-ldap/server/sync.js:461:6)
2021-01-27T12:52:50Z wekan.wekan[237285]:     at runWithEnvironment (packages/meteor.js:1286:24)
2021-01-27T12:52:51Z wekan.wekan[237285]: Exception in callback of async function: TypeError: str.toLowerCase is not a function
2021-01-27T12:52:51Z wekan.wekan[237285]:     at Object.later.parse.text (/snap/wekan/1296/programs/server/npm/node_modules/meteor/percolate_synced-cron/node_modules/later/later.js:1506:34)
2021-01-27T12:52:51Z wekan.wekan[237285]:     at Object.schedule (packages/wekan-ldap/server/sync.js:452:22)
2021-01-27T12:52:51Z wekan.wekan[237285]:     at scheduleEntry (packages/percolate_synced-cron.js:126:24)
2021-01-27T12:52:51Z wekan.wekan[237285]:     at packages/percolate_synced-cron.js:163:7
2021-01-27T12:52:51Z wekan.wekan[237285]:     at Function._.each._.forEach (packages/underscore.js:147:22)
2021-01-27T12:52:51Z wekan.wekan[237285]:     at packages/percolate_synced-cron.js:162:7
2021-01-27T12:52:51Z wekan.wekan[237285]:     at Object.startup (packages/meteor.js:992:5)
2021-01-27T12:52:51Z wekan.wekan[237285]:     at Object.SyncedCron.start (packages/percolate_synced-cron.js:160:10)
2021-01-27T12:52:51Z wekan.wekan[237285]:     at notificationCleanupDebounced (models/users.js:1119:16)
2021-01-27T12:52:51Z wekan.wekan[237285]:     at runWithEnvironment (packages/meteor.js:1286:24)
```

Thank you very much and best regards,
Andi

--- a/packages/wekan-ldap/server/ldap.js
+++ b/packages/wekan-ldap/server/ldap.js
@@ -19,7 +19,7 @@ export default class LDAP {
       idle_timeout                       : this.constructor.settings_get('LDAP_IDLE_TIMEOUT'),
       encryption                         : this.constructor.settings_get('LDAP_ENCRYPTION'),
       ca_cert                            : this.constructor.settings_get('LDAP_CA_CERT'),
-      reject_unauthorized                : this.constructor.settings_get('LDAP_REJECT_UNAUTHORIZED') || true,
+      reject_unauthorized                : this.constructor.settings_get('LDAP_REJECT_UNAUTHORIZED') !== undefined ? this.constructor.settings_get('LDAP_REJECT_UNAUTHORIZED') : true,
       Authentication                     : this.constructor.settings_get('LDAP_AUTHENTIFICATION'),
       Authentication_UserDN              : this.constructor.settings_get('LDAP_AUTHENTIFICATION_USERDN'),
       Authentication_Password            : this.constructor.settings_get('LDAP_AUTHENTIFICATION_PASSWORD'),

